wfsim/strax_interface.py

```diff
@@ -60,34 +65,46 @@ def rand_instructions(c):
     return instructions
 
 
-@export
-def read_optical(file):
-    data = uproot.open(file)
-    all_ttrees = dict(data.allitems(filterclass=lambda cls: issubclass(cls, uproot.tree.TTreeMethods)))
-    e = all_ttrees[next(iter(all_ttrees))]
 
-    n_events = len(e.array('eventid'))
+def read_optical(file, c):
+
+    data = uproot.open(file)
+    if c['mc_version_above_4']:
+        e = data.get('events')
+    else:
+        all_ttrees = dict(data.allitems(filterclass=lambda cls: issubclass(cls, uproot.tree.TTreeMethods)))
+        e = all_ttrees[next(iter(all_ttrees))]
```

Comment:
See these lines doing a similar job: https://github.com/XENONnT/epix/blob/master/epix/io.py#L91-L102 You wouldn't need the `mc_version_above_4` arg in that case.

Reply:
There is still a change in offset of nveto pmts

Reply:
Uhm, but that version was in fact buggy due to the PMT DAQ naming issue, so this is probably a reason to just not use nVeto sims before MC v4. But okay, just stick to what @mzks and you agree on.

Reply:
Oke then I'll change it to throw an exception when someone is using the old mc